expyfun/io/_parse.py

```diff
@@ -73,3 +99,118 @@ def read_tab(fname, group_start='trial_id', group_end='trial_ok'):
+        stop_str = 'tracker_' + str(tracker_id) + '_stop'
+        tracker_stop_idx = np.where([r[1] == stop_str for r in raw])[0]
+        if len(tracker_stop_idx) == 0:
+            raise ValueError('Tracker {} has not stopped. All Trackers '
+                             'must be stopped.'.format(tracker_id))
+        responses = json.loads(raw[tracker_stop_idx[0]][2])['responses']
+        # feed in responses from tracker_ID_stop
+        for r in responses:
+            tr[-1].respond(r)
+    return tr
+
+
+def reconstruct_dealer(fname):
+    """Reconstruct TrackerDealer object from .tab files.
+
+    The ``reconstruct_tracker`` function will be called to retrieve the
+    trackers.
+
+    Parameters
+    ----------
+    fname : str
+        Input filename.
+
+    Returns
+    -------
+    dealer : list of TrackerDealer
+        The TrackerDealer objects with all responses such that they are in
+        their stopped state. If only one dealer is found in the file, it will
+        still be stored in a list and will be assessible as ``td[0]``.
+    """
+    from ..stimuli import TrackerDealer
+    raw = read_tab_raw(fname)
+
+    # find infor on dealer
+    dealer_idx = np.where([r[1] == 'dealer_identify' for r in raw])[0]
+    if len(dealer_idx) == 0:
+        raise ValueError('There are no TrackerDealers in this file.')
+    dealer = []
+    for ii in dealer_idx:
+        dealer_id = ast.literal_eval(raw[ii][2])['dealer_id']
+        dealer_init_str = 'dealer_' + str(dealer_id) + '_init'
+        dealer_dict_idx = np.where([r[1] == dealer_init_str
+                                    for r in raw])[0][0]
+        dealer_dict = ast.literal_eval(raw[dealer_dict_idx][2])
+        dealer_trackers = dealer_dict['trackers']
+
+        # match up tracker objects to id
+        trackers = reconstruct_tracker(fname)
+        tr_objects = []
+        for t in dealer_trackers:
+            idx = np.where([t == t_id._tracker_id for t_id in trackers])[0][0]
+            tr_objects.append(trackers[idx])
+
+        # make the dealer object
+        max_lag = dealer_dict['max_lag']
+        pace_rule = dealer_dict['pace_rule']
+        dealer.append(TrackerDealer(None, tr_objects, max_lag, pace_rule))
+
+        # force input responses/log data
+        dealer_stop_str = 'dealer_' + str(dealer_id) + '_stop'
+        dealer_stop_idx = np.where([r[1] == dealer_stop_str for r in raw])[0]
+        if len(dealer_stop_idx) == 0:
+            raise ValueError('TrackerDealer {} has not stopped. All dealers '
+                             'must be stopped.'.format(dealer_id))
+        dealer_stop_log = json.loads(raw[dealer_stop_idx[0]][2])
+
+        shape = dealer_dict['shape']
+        log_response_history = dealer_stop_log['response_history']
+        log_x_history = dealer_stop_log['x_history']
+        log_tracker_history = dealer_stop_log['tracker_history']
+
+        dealer[-1]._shape = shape
+        dealer[-1]._trackers.reshape(shape)
```

Comment:
Is this shape tested anywhere? I don't think it operates inplace:

https://docs.scipy.org/doc/numpy-1.12.0/reference/generated/numpy.ndarray.reshape.html

You probably want `dealer[-1]._trackers.shape = shape`.